controllers/userControllers.js

```diff
@@ -0,0 +1,111 @@
+const User = require("../models/User");
+const bcryptjs = require("bcryptjs");
+
+/**
+ * things to think about
+ * how we grab users : through req.body or req.params
+ *
+ */
+
+exports.getUsers = async (req, res) => {
+  let users = await User.find();
+  res.json(users);
+};
+
+exports.getUser = async (req, res, next) => {
+  // Grab the id from the url
+  const { id } = req.params;
+  // Find the todo with that id
+  try {
+    let user = await User.findById(id);
+    res.json(user);
+  } catch (err) {
+    //todo : are we making custom errors?
+    next(err); // found error of not found todo
+  }
+};
+
+exports.updateUser = async (req, res) => {
+  // Grab the id from the url
+  const { id } = req.params;
+  // todo : do we have a middle ware to check the req.body to see if the information is good or allowed?
+  try {
+    let userUpdated = await User.findByIdAndUpdate(id, req.body, {
+      new: true,
+    });
```

Comment:
if we going with a presave hook then findByIdAndUpdate will not work

Reply:
presave hook?

Reply:
analog to the todo app we did in class, if we us bcrypt saving .

Reply:
` let user = await User.findById(id);
    // update the user fields
    Object.assign(user, req.body);
    const userUpdated = await user.save(); // => this will trigger the pre save hook
    res.json(userUpdated);`

Reply:
also done :)